prog/dftb+/lib_dftbplus/main.F90

```diff
@@ -6247,6 +6247,80 @@ subroutine getReksInitialSettings(env, denseDesc, h0, over, neighbourList, &
   end subroutine getReksInitialSettings
 
 
+  !> Normalize eigenvectors with unitary transformation
+  subroutine renormalizeEigenvecs(env, electronicSolver, eigvecsReal, reks)
+
+    use dftbp_blasroutines, only : gemm
+
+    !> Environment settings
+    type(TEnvironment), intent(inout) :: env
+
+    !> Electronic solver information
+    type(TElectronicSolver), intent(inout) :: electronicSolver
+
+    !> Eigenvectors on eixt
+    real(dp), intent(inout) :: eigvecsReal(:,:,:)
+
+    !> data type for REKS
+    type(TReksCalc), intent(inout) :: reks
+
+    real(dp), allocatable :: tmpMat(:,:)
+    real(dp), allocatable :: tmpS(:,:)
+    real(dp), allocatable :: tmpC(:,:)
+    real(dp), allocatable :: tmpEigen(:)
+    real(dp), allocatable :: unitaryMat(:,:)
+
+    integer :: nOrb, ii
+
+    nOrb = size(reks%overSqr,dim=1)
+
+    allocate(tmpMat(nOrb,nOrb))
+    allocate(tmpS(nOrb,nOrb))
+    allocate(tmpC(nOrb,nOrb))
+    allocate(tmpEigen(nOrb))
+    allocate(unitaryMat(nOrb,nOrb))
+
+    ! Calculate CSC = C^T_old * S_new * C_old
+    tmpMat(:,:) = 0.0_dp
+    call gemm(tmpMat, eigvecsReal(:,:,1), reks%overSqr, transA='T')
+    tmpC(:,:) = 0.0_dp
+    call gemm(tmpC, tmpMat, eigvecsReal(:,:,1))
+
+    ! Make an identity matrix
+    tmpS(:,:) = 0.0_dp
+    do ii = 1, nOrb
+      tmpS(ii,ii) = 1.0_dp
+    end do
+
+    ! Diagonalize CSC to obtain a unitary matrix, U = CSC^(-1/2)
+
+    tmpEigen(:) = 0.0_dp
+    call env%globalTimer%startTimer(globalTimers%diagonalization)
+    ! tmpC becomes eigenvectors (X) of CSC
+    call diagDenseMtx(electronicSolver, 'V', tmpC, tmpS, tmpEigen)
```

Comment:
Any reason to use a generalized eigensolver if tmpS = I ?

Reply:
Is there a specific eigensolver to diagonalize certain matrix without tmpS? If then, `solveSecularEqn` routine also can be modified.

Reply:
In the dftbp_eigensolver module, heev wraps the ?syev and ?heev symetric/hermitian solvers.

Reply:
Now REKS uses heev routine for diagonalization without an identity matrix. Please see the commit 050577b.